crates/bevy_render/src/texture/hdr_texture_loader.rs

```diff
@@ -46,10 +51,14 @@ impl AssetLoader for HdrTextureLoader {
         reader.read_to_end(&mut bytes).await?;
         let decoder = image::codecs::hdr::HdrDecoder::new(bytes.as_slice())?;
         let info = decoder.metadata();
-        let rgb_data = decoder.read_image_hdr()?;
-        let mut rgba_data = Vec::with_capacity(rgb_data.len() * format.pixel_size());
+        let total_bytes = decoder.total_bytes().try_into()?;
+        let dynamic_image = DynamicImage::from_decoder(decoder)?;
+        let image_buffer = dynamic_image
+            .as_rgb32f()
+            .ok_or(HdrTextureLoaderError::ImageReferenceRgb32F)?;
+        let mut rgba_data = Vec::with_capacity(total_bytes);
```

Comment:
This currently isn't allocating enough space for the alpha channel, so it has to be resized later.

Maybe `image_buffer.pixels().len() * format.pixel_size()`, but you can check that the length of the resulting vec is the same as the initial capacity you specified to be sure.